test/e2e/storage/vsphere/vsphere_zone_support.go

```diff
@@ -0,0 +1,378 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package vsphere
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/kubernetes/test/e2e/framework"
+	"k8s.io/kubernetes/test/e2e/storage/utils"
+)
+
+const (
+	VsanDatastore1  = "vsanDatastore"
```

Comment:
Everything looks good to me, except this test requires specific names for datastores, policies and zones on the testbed.

It would be good, if we can read these variables from environment, instead of keeping them hard coded.
https://github.com/kubernetes/kubernetes/blob/master/test/e2e/storage/vsphere/vsphere_common.go#L55:6

Reply:
Makes sense. Fixed the code to read the variables from the environment.